python/llm/src/bigdl/llm/transformers/models/chatglm2.py

```diff
@@ -246,13 +246,13 @@ def chatglm2_attention_forward_8eb45c(
         key_layer = key_layer.permute(1, 2, 0, 3).unsqueeze(-3)  # [bs, nh/k, sl, hn]
         key_layer = key_layer.expand(-1, -1, query_group_size, -1, -1)
         key_layer = key_layer.contiguous().view((batch_size,
```

Comment:
Looks like the `key_layer` here is of shape `[batch_size, num_multi_query_groups_per_partition, query_group_size, sequence_length, hidden_size_per_attention_head]`, how can it be viewed as `[batch_size, num_multi_query_groups_per_partition, sequene_length, hidden_size_per_attention_head]` if `query_group_size` is not 1.